tests/next_tests/unit_tests/iterator_tests/test_type_inference.py

```diff
@@ -18,6 +18,31 @@
 from gt4py.next.iterator import ir, ir_makers as im, type_inference as ti
 
 
+def test_unsatisfiable_constraints():
+    a = ir.Sym(id="a", dtype=("float32", False))
+    b = ir.Sym(id="b", dtype=("int32", False))
+
+    testee = im.lambda_(a, b)(im.plus("a", "b"))
+
+    # TODO(tehrengruber): For whatever reason the ordering in the error message is not
+    #  deterministic. Ignoring for now, as we want to refactor the type inference anyway.
```

Comment:
Maybe update the comment to reflect why we think the ordering is not deterministic.